text/text.go

```diff
@@ -0,0 +1,197 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package text
+
+import (
+	"image/color"
+	"math"
+
+	"gonum.org/v1/plot/font"
+	"gonum.org/v1/plot/vg"
+)
+
+// Handler parses, formats and renders text.
+type Handler interface {
+	// Extents returns the FontExtents for a font.
+	Extents(fnt font.Font) font.FontExtents
+
+	// Lines splits a given block of text into separate lines.
+	Lines(txt string) []string
+
+	// Box returns the bounding box of the given non-multiline text where:
+	//  - width is the horizontal space from the origin.
+	//  - height is the vertical space above the baseline.
+	//  - depth is the vertical space below the baseline, a positive number.
+	Box(txt string, fnt font.Font) (width, height, depth vg.Length)
+
+	// Draw renders the given text with the provided style and position
+	// on the canvas.
+	Draw(c vg.Canvas, txt string, sty Style, pt vg.Point)
+}
+
+// XAlignment specifies text alignment in the X direction. Three preset
+// options are available, but an arbitrary alignment
+// can also be specified using XAlignment(desired number).
+type XAlignment float64
+
+const (
+	// XLeft aligns the left edge of the text with the specified location.
+	XLeft XAlignment = 0
+	// XCenter aligns the horizontal center of the text with the specified location.
+	XCenter XAlignment = -0.5
+	// XRight aligns the right edge of the text with the specified location.
+	XRight XAlignment = -1
+)
+
+// YAlignment specifies text alignment in the Y direction. Three preset
+// options are available, but an arbitrary alignment
+// can also be specified using YAlignment(desired number).
+type YAlignment float64
+
+const (
+	// YTop aligns the top of of the text with the specified location.
+	YTop YAlignment = -1
+	// YCenter aligns the vertical center of the text with the specified location.
+	YCenter YAlignment = -0.5
+	// YBottom aligns the bottom of the text with the specified location.
+	YBottom YAlignment = 0
+)
+
+// Position specifies the text position.
+const (
+	PosLeft   = -1
+	PosBottom = -1
+	PosCenter = 0
+	PosTop    = +1
+	PosRight  = +1
+)
+
+// Style describes what text will look like.
+type Style struct {
+	// Color is the text color.
+	Color color.Color
+
+	// Font is the font description.
+	Font font.Font
+
+	// Rotation is the text rotation in radians, performed around the axis
+	// defined by XAlign and YAlign.
+	Rotation float64
+
+	// XAlign and YAlign specify the alignment of the text.
+	XAlign XAlignment
+	YAlign YAlignment
+
+	// Handler parses and formats text according to a given
+	// dialect (Markdown, LaTeX, plain, ...)
+	// The default is a plain text handler.
+	Handler Handler
+}
+
+// FontExtents returns the FontExtents of this Style's font.
+func (sty Style) FontExtents() font.FontExtents {
```

Comment:
Though I think this would stay as `FontExtents`; 

```
// FontExtents returns the font.Extents of this Style's font.
func (sty Style) FontExtents() font.Extents
```